File: backend/app/models/education.py
from __future__ import annotations

from sqlalchemy import JSON, Boolean, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base, PublishableMixin, SoftDeleteMixin, TimestampMixin


class Education(TimestampMixin, PublishableMixin, SoftDeleteMixin, Base):
    __tablename__ = "education"

    id: Mapped[int] = mapped_column(primary_key=True)
    institution: Mapped[str] = mapped_column(String(200), nullable=False)
    degree_en: Mapped[str] = mapped_column(String(200), nullable=False)
    degree_it: Mapped[str] = mapped_column(String(200), nullable=False)
    field: Mapped[str | None] = mapped_column(String(200), nullable=True)
    location: Mapped[str] = mapped_column(String(120), nullable=False)
    start_year: Mapped[int | None] = mapped_column(Integer, nullable=True)
    end_year: Mapped[int | None] = mapped_column(Integer, nullable=True)
    is_ongoing: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    description_en: Mapped[str | None] = mapped_column(Text, nullable=True)
    description_it: Mapped[str | None] = mapped_column(Text, nullable=True)
    activities: Mapped[list[str]] = mapped_column(JSON, nullable=False, default=list)
    url: Mapped[str | None] = mapped_column(String(255), nullable=True)
    logo_media_url: Mapped[str | None] = mapped_column(String(255), nullable=True)
    sort_order: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    internal_notes: Mapped[str | None] = mapped_column(Text, nullable=True)

File: frontend/src/admin/types/education.ts
export interface EducationListItem {
  id: number;
  institution: string;
  degree_en: string;
  sort_order: number;
  publication_status: "DRAFT" | "SCHEDULED" | "PUBLISHED" | "ARCHIVED";
  deleted_at: string | null;
}

export interface EducationAdminDetail {
  id: number;
  institution: string;
  degree_en: string;
  degree_it: string;
  field: string | null;
  location: string;
  start_year: number | null;
  end_year: number | null;
  is_ongoing: boolean;
  description_en: string | null;
  description_it: string | null;
  activities: string[];
  url: string | null;
  logo_media_url: string | null;
  sort_order: number;
  internal_notes: string | null;
  publication_status: "DRAFT" | "SCHEDULED" | "PUBLISHED" | "ARCHIVED";
  publish_at: string | null;
  unpublish_at: string | null;
  published_at: string | null;
  deleted_at: string | null;
  created_at: string;
  updated_at: string;
}

export type EducationWritePayload = Omit<
  EducationAdminDetail,
  | "id"
  | "publication_status"
  | "publish_at"
  | "unpublish_at"
  | "published_at"
  | "deleted_at"
  | "created_at"
  | "updated_at"
>;

export interface EducationRevisionSummary {
  id: number;
  schema_version: number;
  snapshot: Record<string, unknown>;
  created_at: string;
  created_by_email: string | null;
}
